# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1632859768 -7200
# Parent  7d2fb9deb292f1acb38db43f8a57862ecb58bcd5
Feature: Fix background colour of default buttons to be less surprising.

diff --git a/mail/themes/shared/mail/themeableDialog.css b/mail/themes/shared/mail/themeableDialog.css
--- a/mail/themes/shared/mail/themeableDialog.css
+++ b/mail/themes/shared/mail/themeableDialog.css
@@ -18,51 +18,52 @@
   --checkbox-unchecked-hover-bgcolor: var(--field-background-color);
   --field-text-color: FieldText;
   --field-background-color: Field;
   --field-border-color: rgba(128, 128, 128, .6);
   --field-border-hover-color: rgba(128, 128, 128, .8);
   --highlight-color: Highlight;
   --popup-item-hover: rgba(128, 128, 128, .2);
   --popup-item-hover-text: MenuText;
-  --primary-text-color: HighlightText;
-  --primary-background-hover: color-mix(in srgb, var(--highlight-color) 85%, black);
-  --primary-background-active: color-mix(in srgb, var(--highlight-color) 78%, black);
+  --primary-background: rgba(255, 255, 255, 0.4);
+  --primary-background-hover: color-mix(in srgb, var(--primary-background) 85%, black);
+  --primary-background-active: color-mix(in srgb, var(--primary-background) 78%, black);
   --primary-focus-border: -moz-Dialog;
   --richlist-button-background: -moz-Dialog;
   --tab-hover-background: hsla(0, 0%, 50%, 0.15);
   --tab-selected-background: hsla(0, 0%, 50%, 0.25);
 }
 
 :root:-moz-lwtheme {
   --box-text-color: #0c0c0d;
   --box-background-color: #fff;
   --box-border-color: #adadae;
   --field-text-color: #181920;
   --field-background-color: #fff;
   --field-border-color: rgba(0, 0, 0, 0.3);
   --field-border-hover-color: rgba(0, 0, 0, 0.4);
   --popup-item-hover: rgba(69, 161, 255, 0.2);
   --popup-item-hover-text: var(--box-text-color);
+  --primary-background: color-mix(in srgb, var(--button-bgcolor) 30%, white);
   --primary-focus-border: var(--lwt-accent-color);
   background-color: var(--lwt-accent-color);
   color: var(--lwt-text-color);
 }
 
 :root:-moz-lwtheme-brighttext {
   --box-text-color: rgb(249, 249, 250);
   --box-background-color: #202023;
   --box-border-color: rgba(249, 249, 250, 0.3);
   --field-text-color: #f9f9fa;
   --field-background-color: #353537;
   --field-border-color: hsla(0, 0%, 70%, 0.4);
   --field-border-hover-color: hsla(0, 0%, 70%, 0.5);
   --highlight-color: #0a84ff;
   --highlight-background: #fff;
-  --primary-text-color: #f9f9fa;
+  --primary-background: color-mix(in srgb, var(--button-bgcolor) 78%, white);
   --tab-hover-background: hsla(0, 0%, 50%, 0.3);
   --tab-selected-background: hsla(0, 0%, 50%, 0.5);
 }
 
 :root[lwt-tree] {
   --richlist-button-background: var(--sidebar-background-color);
 }
 
@@ -247,23 +248,22 @@ html|input[type="color"]:not([disabled="
 button[open],
 button[open]:hover,
 button:not([disabled="true"]):hover:active,
 html|input[type="color"]:not([disabled="true"]):hover:active {
   background-color: var(--button-active-bgcolor);
 }
 
 button[default] {
-  background-color: var(--highlight-color);
-  color: var(--primary-text-color) !important;
+  background-color: var(--primary-background);
 }
 
-button[default]:-moz-focusring {
-  border-color: var(--primary-focus-border);
-  outline-offset: 0;
+/* Note that [default] and [default="true"] are two different things */
+button[default="true"] {
+  border-color: var(--highlight-color);
 }
 
 button[default]:not([disabled="true"]):hover {
   background-color: var(--primary-background-hover);
 }
 
 button[default]:not([disabled="true"]):hover:active {
   background-color: var(--primary-background-active);
